multik-jvm/src/main/kotlin/org/jetbrains/kotlinx/multik/jvm/pludecomposition.kt

```diff
@@ -0,0 +1,259 @@
+package org.jetbrains.kotlinx.multik.jvm
+
+import org.jetbrains.kotlinx.multik.api.*
+import org.jetbrains.kotlinx.multik.ndarray.data.*
+import org.jetbrains.kotlinx.multik.ndarray.data.set
+import kotlin.math.abs
+import kotlin.math.min
+
+/**
+ * computes alpha * a * b + beta * c and stores result is c
+ * @param alpha scalar
+ * @param beta scalar
+ * @param a matrix
+ * @param b matrix
+ * @param c matrix
+ *
+ * blas / lapack: dgemm
+ */
+private fun dotThenPlusInplace(a: D2Array<Double>, b: D2Array<Double>, c: D2Array<Double>, alpha: Double, beta: Double) {
```

Comment:
is there a need to use inplace multiplication? can not use the already written dot?